src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1030,53 +1034,71 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
                 MethodSymbol overriddenMethod,
                 MethodSymbol overridingMethod,
                 DiagnosticBag diagnostics,
+                bool checkReturnType,
                 bool checkParameters)
             {
-                var compilation = overridingMethod.DeclaringCompilation;
-                if (overriddenMethod is null ||
-                    overridingMethod is null ||
-                    compilation is null ||
-                    !compilation.IsFeatureEnabled(MessageID.IDS_FeatureNullableReferenceTypes))
-                {
-                    // Don't do any validation if the nullable feature is not enabled or
-                    // the override is not written directly in source
-                    return;
-                }
+                CheckValidNullableMethodOverride(overridingMethod.DeclaringCompilation, overriddenMethod, overridingMethod, diagnostics,
+                                                 checkReturnType ?
+                                                    (diagnostics, overriddenMethod, overridingMethod, location) => diagnostics.Add(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride, location) :
+                                                    (Action<DiagnosticBag, MethodSymbol, MethodSymbol, Location>)null,
+                                                 checkParameters ?
+                                                     (diagnostics, overriddenMethod, overridingMethod, overridingParameter, location) =>
```

Comment:
>(diagnostics, overriddenMethod, overridingMethod, overridingParameter, location) => [](start = 53, length = 83)

nit: consider using a local function for those funcs in various invocations of `CheckValidNullableEventOverride`

Reply:
> nit: consider using a local function for those funcs in various invocations of CheckValidNullableEventOverride

Lambdas are used intentionally to take advantage of automatic caching.

---
In reply to: [296389099](https://github.com/dotnet/roslyn/pull/36663#discussion_r296389099) [](ancestors = 296389099)